### PR TITLE
Update configuration to be compatible with current v1alpha2 spec

### PR DESCRIPTION
**Pull request by hasheddan**
Modifies the manifests to make this sample compliant with the v1alpha2 spec of OAM and the Crossplane implementation. It preserves the ability to run postgres in-cluster but adds additional support for consuming a managed database from any cloud provider.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### CODE REVIEW

**Comment by hasheddan** [comment]
cc @artursouza (it won't let me add reviewers)

Will have update to `README.md` coming shortly :+1:

File: 2.ServiceTracker_App/oam-remote.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: oam
+---
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: addon-oam-kubernetes-remote
+  namespace: oam
+spec:
+  package: "crossplane/addon-oam-kubernetes-remote:master"
```
Comment (wonderflow) [inline 2.ServiceTracker_App/oam-remote.yaml:12]: could we use local stack and make it as default so we won't need this yaml file.

Comment (hasheddan) [reply 2.ServiceTracker_App/oam-remote.yaml:12]: > mark as default

Do you mean assume that the user already has the controller running? It seems to make sense to have an `oam-remote.yaml` and `oam-local.yaml` file here and include instructions for both in the `README.md`.

Comment (wonderflow) [reply 2.ServiceTracker_App/oam-remote.yaml:12]: Yeah, I assume `oam-local` along with crossplane are all default installed together.

Comment (hasheddan) [reply 2.ServiceTracker_App/oam-remote.yaml:12]: Hmm I don't know why we wouldn't provide instruction here for both cases. There currently is not documentation anywhere else for installing the remote and local addons so I think it makes sense to provide here.

Comment (hasheddan) [reply 2.ServiceTracker_App/oam-remote.yaml]: @wonderflow I removed this in favor of just mentioning that either the local or remote addon must be enabled :+1:

File: 2.ServiceTracker_App/Components/tracker-data-component.yaml [hunk 2]
```diff
@@ -69,31 +68,30 @@ spec:
...
-      type: string
-      required: false
-    - name: dbpassword
-      description: database password
-      type: string
+    - name: dbsecret
+      description: secret with database connection information
       required: false
+      fieldPaths:
+      - spec.containers[0].env[0].fromSecret.name
```
Comment (hongchaodeng) [inline 2.ServiceTracker_App/Components/tracker-data-component.yaml:75]: What does it mean to have three fieldPaths?

Comment (hasheddan) [reply 2.ServiceTracker_App/Components/tracker-data-component.yaml:75]: It means that the value passed for this parameter will populate each of the fields in the array of field paths :+1:

Comment (artursouza) [reply 2.ServiceTracker_App/Components/tracker-data-component.yaml:75]: I am not a fan of this solution to pass in param values but that is what the v1alpha2 spec says.

**Comment by hongchaodeng** [comment]
lgtm.
Thanks @hasheddan !

**Pull request merged**